### ISSUE TITLE
BranchMetricsKit initializes Branch session twice in a simple app startup scenario

### ISSUE THREAD

**Issue by afedorychak** [mParticle/mparticle-android-sdk#54]
Using mParticle 5.14.10.

This happens when the user simply starts the app. Below is the sequence of events and description of our setup.

MParticle is started in app's onCreate. We're setting it up with an MParticle's AttributionListener to be notified about resolved deep links.

Our DeepLinkActivity receives the intent and decides (somewhere after its onCreate) that it needs to start a HomeActivity because it's not a deeplink (DeepLinkActivity acts as a dispatcher for any entry into the app).

BranchMetricsKit initializes branch session in BranchMetricsKit.onKitCreate.

HomeActivity eventually becomes resumed and this triggers "app in foreground" for BranchMetricsKit which then AGAIN tries to initialize session in BranchMetricsKit.onApplicationForeground.

Branch.initializeSession doesn't like this at line 2016 in `/Users/<my home dir>/.gradle/caches/modules-2/files-2.1/io.branch.sdk.android/library/5.0.1/daf17ab6cc659fa836a407d5122bae35932d01c8/library-5.0.1-sources.jar!/io/branch/referral/Branch.java` and this causes onError call to the attribution listener which polluted our logs and error monitoring.

Here's the full time line of events (the first number is SystemClock.uptimeMillis() truncated for brevity sake):

0081 -- DeepLinkActivity routes the intent in onCreate
0344 -- BranchMetricsKit.onKitCreate
0417 -- Brach.initSession: getInitState()=UNINITIALISED, forceBranchSession=false callback=com.mparticle.kits.BranchMetricsKit@919ee4e
0751 -- HomeActivity onCreate
1652 -- HomeActivity onResume
1796 -- BranchMetricsKit.onApplicationForeground
1846 -- Brach.initSession: getInitState()=INITIALISING, forceBranchSession=false callback=com.mparticle.kits.BranchMetricsKit@919ee4e (at this point we receive AttributionListener.onError)
3248 -- DeepLinkActivity onDestroy

**Comment by willpassidomo** [comment]
@afedorychak hey thanks for reporting this issue. I'm sorry for the lag in responding, I just returned from pat leave and I did not cover checking these issue.

It looks like there was some code added recently to initialize a session on `onApplicationForeground()` which is causing the session to init a second time in the scenario you described. I'm going to contact the Branch devs who added it and take a look at what's going on
